### PR TITLE
add Travis-CI to the repo

### PR DESCRIPTION
**Pull request by zimbatm**
todo:
* [ ] setup Travis-CI on the repo
* [ ] test that it works

### CODE REVIEW

**Comment by yegortimoshenko** [comment]
I don't think it would work because this test needs NixOS, and I don't think Travis CI uses NixOS (they are likely to use something like Debian with Nix installed).

File: tests/eval-test.sh [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
 for profile in `${find[@]}`; do
-    echo evaluating $profile >&2
+  echo evaluating $profile >&2
 
-    nixos-rebuild \
-	-I nixos-config=tests/eval-test.nix \
-	-I nixos-hardware-profile=$profile \
-	dry-build
-
-    if [ $? -ne 0 ]; then
```
Comment (yegortimoshenko) [inline tests/eval-test.sh]: Without this, test would never fail.

Comment (zimbatm) [reply tests/eval-test.sh]: `set -e` takes care of this

**Comment by zimbatm** [comment]
fixed the test script so that it works on any system that has nix installed

**Comment by Mic92** [comment]
In doubt, it is possible to create VMs on non-nixos systems: `nix-build '<nixpkgs/nixos>' -A vm -k -I nixos-config=./vm.nix`

File: tests/eval-test.sh (hunk 1 above)
Comment (yegortimoshenko) [reply tests/eval-test.sh]: Great! Didn't know about `set -e`.

**Comment by yegortimoshenko** [comment]
I think shell scripts should be portable. Also, changing evaluation rules makes it more complex to reason about. Let's leave it at `#!/bin/sh`, `set -e`, and `find`.

**Comment by yegortimoshenko** [comment]
Sorry for force-pushing on top of your work, for reference, it's in fc032be7355ef8d1226fa77f0ef3a0c153e36100. Again, I think it makes the script more complex and I don't see any point in making it less verbose.

**Comment by zimbatm** [comment]
@yegortimoshenko it's fine to disagree but please stop merging things into master while we are having a discussion. Anyways, follow-up in https://github.com/NixOS/nixos-hardware/pull/44

**Comment by yegortimoshenko** [comment]
@zimbatm I've only reverted your pull request to a previous state. The point of this pull request seemingly was to make Travis CI work and it works, so I've merged it.

**Pull request merged**